server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java

```diff
@@ -748,14 +748,16 @@ public void testIndexHasDuplicateData() throws IOException {
                 LongPoint.encodeDimension(value, longBytes, 0);
                 w.add(longBytes, docId);
             }
-            long indexFP;
-            try (IndexOutput out = dir.createOutput("bkd", IOContext.DEFAULT)) {
-                indexFP = w.finish(out);
+            try (IndexOutput metaout = dir.createOutput("bkdmeta", IOContext.DEFAULT);
+                 IndexOutput indexout = dir.createOutput("bkdindex", IOContext.DEFAULT);
+                 IndexOutput dataout = dir.createOutput("bkddata", IOContext.DEFAULT)) {
+                w.finish(metaout, indexout, dataout).run();
```

Comment:
You need to change the finish method to something like:

```
Runnable finalizer = writer.finish(metaOut, indexOut, dataOut);
      if (finalizer != null) {
        metaOut.writeInt(fieldInfo.number);
        finalizer.run();
      }
```

Reply:
Or even better like this:

```    
        Runnable finalizer = w.finish(out, out, out);
        long indexFP = out.getFilePointer();
        finalizer.run();
 ```

Then when you call the reader you need to make sure retain is in the right position:

```
metain.seek(indexFP);
```